src/mono/mono/sgen/sgen-scan-object.h

```diff
@@ -14,7 +14,7 @@
  * "start" will point to the start of the next object, if the scanned
  * object contained references.  If not, the value of "start" should
  * be considered undefined after executing this code.  The object's
- * GC descriptor must be in the variable "mword desc".
+ * GC descriptor must be in the variable "SgenDescriptor desc".
```

Comment:
These are different type depending on host/target, mword is host machine word while SgenDescriptor is target machine word, have you validated that calls only uses SgenDescriptor and we don't end up casting mword to SgenDescriptor or SgenDescriptor to mword since that can lead to data loss depending on host/target combinations.

Reply:
Yep I checked all the callsites and they are all called with SgenDescriptor type without casting.